src/aleph/vm/controllers/configuration.py

```diff
@@ -0,0 +1,47 @@
+from enum import Enum
+from pathlib import Path
+from typing import Optional, Union
+
+from pydantic import BaseModel
+
+from aleph.vm.conf import Settings, settings
+
+
+class VMConfiguration(BaseModel):
+    use_jailer: bool
+    firecracker_bin_path: Path
+    jailer_bin_path: Path
+    config_file_path: Path
+    init_timeout: float
+
+
+class QemuVMConfiguration(BaseModel):
+    qemu_bin_path: str
+    cloud_init_drive_path: Optional[str]
+    image_path: str
+    monitor_socket_path: Path
+    qmp_socket_path: Path
+    vcpu_count: int
+    mem_size_mb: int
+    interface_name: Optional[str]
+
+
+class HypervisorType(str, Enum):
+    qemu = "qemu"
+    firecracker = "firecracker"
+
+
+class Configuration(BaseModel):
+    vm_id: int
+    settings: Settings
+    vm_configuration: Union[QemuVMConfiguration, VMConfiguration]
+    hypervisor: HypervisorType = HypervisorType.firecracker
+
+
+def save_controller_configuration(vm_hash: str, configuration: Configuration) -> Path:
+    """Save VM configuration to be used by the controller service"""
+    config_file_path = Path(f"{settings.EXECUTION_ROOT}/{vm_hash}-controller.json")
+    with config_file_path.open("w") as controller_config_file:
+        controller_config_file.write(configuration.json(by_alias=True, exclude_none=True, indent=4))
```

Comment:
When attempting to schedule a program instead of an instance from the scheduler API, I get an error 500.
While this is not expected to be done, it should not crash with an error 500.

```python
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/root/aleph-vm-remote/src/aleph/vm/orchestrator/supervisor.py", line 52, in server_version_middleware
    resp: web.StreamResponse = await handler(request)
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/root/aleph-vm-remote/src/aleph/vm/orchestrator/views/__init__.py", line 290, in update_allocations
    await start_persistent_vm(instance_item_hash, pubsub, pool)
  File "/root/aleph-vm-remote/src/aleph/vm/orchestrator/run.py", line 267, in start_persistent_vm
    execution = await create_vm_execution(vm_hash=vm_hash, pool=pool, persistent=True)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/aleph-vm-remote/src/aleph/vm/orchestrator/run.py", line 60, in create_vm_execution
    execution = await pool.create_a_vm(
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/aleph-vm-remote/src/aleph/vm/pool.py", line 100, in create_a_vm
    await execution.create(vm_id=vm_id, tap_interface=tap_interface)
  File "/root/aleph-vm-remote/src/aleph/vm/models.py", line 218, in create
    await vm.configure()
  File "/root/aleph-vm-remote/src/aleph/vm/controllers/firecracker/program.py", line 327, in configure
    await super().configure()
  File "/root/aleph-vm-remote/src/aleph/vm/controllers/firecracker/executable.py", line 285, in configure
    save_controller_configuration(self.vm_hash, configuration)
  File "/root/aleph-vm-remote/src/aleph/vm/controllers/configuration.py", line 45, in save_controller_configuration
    controller_config_file.write(configuration.json(by_alias=True, exclude_none=True, indent=4))
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/main.py", line 504, in pydantic.main.BaseModel.json
  File "/usr/lib/python3.11/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/usr/lib/python3.11/json/encoder.py", line 202, in encode
    chunks = list(chunks)
             ^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/encoder.py", line 432, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.11/json/encoder.py", line 406, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.11/json/encoder.py", line 406, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.11/json/encoder.py", line 439, in _iterencode
    o = _default(o)
        ^^^^^^^^^^^
  File "pydantic/json.py", line 90, in pydantic.json.pydantic_encoder
TypeError: Object of type 'object' is not JSON serializable
```

Reply:
I already tested it on Debian 11 and 12, but I cannot reproduce that issue.

Reply:
OK, I will reset my test setup and check again.